packages/service-library/tests/test_redis.py

```diff
@@ -153,68 +150,116 @@ async def test_redis_lock_with_ttl(
 
 
 async def test_lock_context(
-    redis_client: Callable[[], RedisClientSDK],
-    faker: Faker,
-    lock_timeout: datetime.timedelta,
+    redis_client_sdk: RedisClientSDK, faker: Faker, lock_timeout: datetime.timedelta
 ):
-    client = redis_client()
     lock_name = faker.pystr()
-    assert await client.is_locked(lock_name) is False
-    async with client.lock_context(lock_name) as ttl_lock:
-        assert await client.is_locked(lock_name) is True
+    assert await _is_locked(redis_client_sdk, lock_name) is False
+    async with redis_client_sdk.lock_context(lock_name) as ttl_lock:
+        assert await _is_locked(redis_client_sdk, lock_name) is True
         assert await ttl_lock.owned() is True
         await asyncio.sleep(5 * lock_timeout.total_seconds())
-        assert await client.is_locked(lock_name) is True
+        assert await _is_locked(redis_client_sdk, lock_name) is True
         assert await ttl_lock.owned() is True
-    assert await client.is_locked(lock_name) is False
+    assert await _is_locked(redis_client_sdk, lock_name) is False
     assert await ttl_lock.owned() is False
 
 
 async def test_lock_context_with_already_locked_lock_raises(
-    redis_client: Callable[[], RedisClientSDK],
-    faker: Faker,
+    redis_client_sdk: RedisClientSDK, faker: Faker
 ):
-    client = redis_client()
     lock_name = faker.pystr()
-    assert await client.is_locked(lock_name) is False
-    async with client.lock_context(lock_name) as lock:
-        assert await client.is_locked(lock_name) is True
+    assert await _is_locked(redis_client_sdk, lock_name) is False
+    async with redis_client_sdk.lock_context(lock_name) as lock:
+        assert await _is_locked(redis_client_sdk, lock_name) is True
 
-        with pytest.raises(AlreadyLockedError):
+        with pytest.raises(CouldNotAcquireLockError):
```

Comment:
AlreadyLockedError is a bit more self-explanatory than CouldNotAcquireLockError no?
or are there new options now?

Reply:
well to be honest I think that if you can detect a timeout then we should take appropriate action. cause the lock might still exist. if there is a timeout you should not decide, this could create huge issues.